src/viewPages/ListArticles.jsx

```diff
@@ -5,33 +5,60 @@ import "../All.css";
 
 export default function ListArticles() {
   const [articles, setArticles] = useState([]);
+  const [filteredArticles, setFilteredArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState(false);
+  const [topic, setTopic] = useState("all");
+  const [topics, setTopics] = useState(["all"]);
 
   useEffect(() => {
     getApi("/api/articles")
       .then((response) => {
         setArticles(response.articles);
+        setFilteredArticles(response.articles);
         setIsLoading(false);
+        // create a list of unique topics from the articles
+        // "Set" is making sure that we only have unique topics and will look like this: {'all', 'topic1', 'topic2', ...}, like an object without keys but with unique values
+        //Array.from(...):  converts the Set back into an array. This is done because Sets are not as easy to work with as with arrays, and the map function is not available for Sets.
+        const uniqueTopics = Array.from(
+          new Set(response.articles.map((article) => article.topic))
```

Comment:
You shouldn't be doing it like this. You should be making a request to the database and sending the topic. However there is a chance you didn't build this functionality into your BE. If you haven't then this is ok.

The problem is that the url is not changing depending on what topic you are on. The brief says you want to be able to send the link to someone.